### DISCUSSION TITLE
Cask

### DISCUSSION THREAD

**Discussion by pjamesnorris** [Homebrew/discussions#5798] [Casks] [answered]
### Output of `brew config`

```shell
brew config
HOMEBREW_VERSION: 4.4.11-2-g03d29a5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 03d29a5743ebd4d1239ad5e68fcf0094766a0f10
Last commit: 4 hours ago
Branch: master
Core tap JSON: 10 Dec 13:27 UTC
Core cask tap JSON: 10 Dec 13:27 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
brew doctor
Error: Unexpected method 'appcast' called on Cask adoptopenjdk12.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
Error: Unexpected method 'appcast' called on Cask adoptopenjdk12.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  imap-uw
```

### Description of issue

What I want to do is uninstall and reinstall httpd, but I keep receiving the **adoptopenjdk12** errors above. I have tried unistalling **adoptopenjdk12** and received the same error messages above, successfully installed **temurin@17** and **temurin**, but... 

Any help would be greatly appreciated!

### Relevant casks

adoptopenjdk12

**Comment by SMillerDev** [comment, marked as answer]
Have you tried `brew uninstall --force adoptopenjdk12`?

**Reply by pjamesnorris** [reply]
@SMillerDev,

No, but it worked! Thank you!!!

But a **brew uninstall httpd** resulted in the error message:

**Uninstalling /opt/homebrew/Cellar/httpd/2.4.62... (1,664 files, 32.2MB)
Error: Could not remove httpd keg! Do so manually:
  sudo rm -rf /opt/homebrew/Cellar/httpd/2.4.62
jnorrisMBA:/opt/homebrew/etc/httpd$ brew uninstall httpd --force
Uninstalling httpd... (1,662 files, 32.1MB)
Error: Could not remove httpd keg! Do so manually:
  sudo rm -rf /opt/homebrew/Cellar/httpd/2.4.62**

I assume because I uninstalled  **adoptopenjdk12**?

So, I did the **sudo rm -rf /opt/homebrew/Cellar/httpd/2.4.62**.

I then:

**brew install httpd**

and was told:

**
==> Auto-updating Homebrew...
Adjust how often this is run with HOMEBREW_AUTO_UPDATE_SECS or disable with
HOMEBREW_NO_AUTO_UPDATE. Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> New Formulae
sfsexp

You have 5 outdated formulae installed.

==> Downloading https://ghcr.io/v2/homebrew/core/httpd/manifests/2.4.62
Already downloaded: /Users/jnorris/Library/Caches/Homebrew/downloads/22cf5fceee1870055fc977c6ee8cfe9cf9ead59565c50648187d4d69c66af1b0--httpd-2.4.62.bottle_manifest.json
==> Fetching httpd
==> Downloading https://ghcr.io/v2/homebrew/core/httpd/blobs/sha256:e07d024239ee944db52ecebb1997c75e15144b343b347788
Already downloaded: /Users/jnorris/Library/Caches/Homebrew/downloads/f1249a3f253bdfa51eeb09dfcd8f37848c7f9f0eb0468698d2d752eb85153728--httpd--2.4.62.arm64_sonoma.bottle.tar.gz
==> Pouring httpd--2.4.62.arm64_sonoma.bottle.tar.gz
==> Caveats
DocumentRoot is /opt/homebrew/var/www.

The default ports have been set in /opt/homebrew/etc/httpd/httpd.conf to 8080 and in
/opt/homebrew/etc/httpd/extra/httpd-ssl.conf to 8443 so that httpd can run without sudo.

To restart httpd after an upgrade:
  brew services restart httpd
Or, if you don't want/need a background service you can just run:
  /opt/homebrew/opt/httpd/bin/httpd -D FOREGROUND
==> Summary
🍺  /opt/homebrew/Cellar/httpd/2.4.62: 1,664 files, 32.2MB
==> Running `brew cleanup httpd`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
**

When I tried:

**brew services start httpd**

I was told:

**Bootstrap failed: 5: Input/output error
Try re-running the command as root for richer errors.
Error: Failure while executing; `/bin/launchctl bootstrap gui/501 /Users/jnorris/Library/LaunchAgents/homebrew.mxcl.httpd.plist` exited with 5.**

So I:

**sudo brew services start httpd**

which resulted in:

**Warning: Taking root:admin ownership of some httpd paths:
  /opt/homebrew/Cellar/httpd/2.4.62/bin
  /opt/homebrew/Cellar/httpd/2.4.62/bin/httpd
  /opt/homebrew/opt/httpd
  /opt/homebrew/opt/httpd/bin
  /opt/homebrew/var/homebrew/linked/httpd
This will require manual removal of these paths using `sudo rm` on
brew upgrade/reinstall/uninstall.
Warning: httpd must be run as non-root to start at user login!
==> Successfully started `httpd` (label: homebrew.mxcl.httpd)**

I then:

**sudo brew services stop httpd**

and

**brew services start httpd**

and was told:

**Bootstrap failed: 5: Input/output error
Try re-running the command as root for richer errors.
Error: Failure while executing; `/bin/launchctl bootstrap gui/501 /Users/jnorris/Library/LaunchAgents/homebrew.mxcl.httpd.plist` exited with 5.**

I take it that there is a permission issue with **/Users/jnorris/Library/LaunchAgents/homebrew.mxcl.httpd.plist**--is that correct? This file's permissions are:

**-rw-r--r--  1 jnorris  staff  777 Dec 10 09:18 /Users/jnorris/Library/LaunchAgents/homebrew.mxcl.httpd.plist**

Are these the correct permissions and, if not, can you tell me what they ought to be?

Thank you very much!
